lib/normalizer.c

```diff
@@ -3488,6 +3488,19 @@ grn_nfkc_normalize_unify_iteration_mark(grn_ctx *ctx,
     return unified_buffer;
   }
 #  undef N_KATAKANA_BYTES
+#  define N_KANJI_BYTES 3
+  else if (current_length == 3 && current[0] == 0xe3 && current[1] == 0x80 &&
+           current[2] == 0xbb && previous_length == N_KANJI_BYTES &&
+           GRN_CHAR_TYPE(grn_nfkc_char_type(previous)) == GRN_CHAR_KANJI) {
+    /* U+303B VERTICAL IDEOGRAPHIC ITERATION MARK */
+    for (size_t i = 0; i < data->previous_length; i++) {
```

Comment:
Is it OK that we use `data->previous_length` here?

Reply:
It's not okay at all. We should have used the `previous_length` here because `data->previous_length` is the current lenght.